allennlp/data/iterators/pass_through_iterator.py

```diff
@@ -0,0 +1,48 @@
+from typing import Iterable, Iterator
+import itertools
+import logging
+
+from allennlp.data.dataset import Batch
+from allennlp.data.instance import Instance
+from allennlp.data.iterators.data_iterator import DataIterator, TensorDict
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DataIterator.register("pass_through")
+class PassThroughIterator(DataIterator):
+    """
+    An iterator which performs no batching or shuffling of instances, only tensorization. E.g,
+    instances are effectively passed 'straight through' the iterator.
+
+    This is essentially the same as a BasicIterator with shuffling disabled, the batch size set
+    to 1, and maximum samples per batch disabled. The only difference is that this iterator
+    removes the batch dimension. This can be useful for rare situations where batching is best
+    performed within the dataset reader (e.g. for contiguous language modeling, or for other
+    problems where state is shared across batches).
+    """
+    def __init__(self):
+        super().__init__(batch_size=1)
+
+    def _create_batches(self, instances: Iterable[Instance], shuffle: bool) -> Iterable[Batch]:
```

Comment:
Keeping an `@overrides` tag here would be nice.  The point is that it makes it obvious to the reader why this method exists.